controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -279,8 +279,17 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 				secret := &corev1.Secret{}
 				secret.Name = resource.Name
 				secret.Namespace = r.storageClassClaim.Namespace
+				_true := true
 				_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, secret, func() error {
-					err := r.own(secret)
+					secret.OwnerReferences = []metav1.OwnerReference{
+						{
+							APIVersion:         r.storageClassClaim.APIVersion,
+							Kind:               r.storageClassClaim.Kind,
+							Name:               r.storageClassClaim.Name,
+							UID:                r.storageClassClaim.UID,
+							BlockOwnerDeletion: &_true,
+						},
+					}
```

Comment:
Re-thinking the subject. I agree with the fact we need a finalizer, mainly because of the need to invoke an API call during delete (revoke call), so no need for the `BlockOwnerDeletion` field.

In addition, even if we wanted to support the field, I would suggest having it an argument on the `own()` call and not inline the owner ref section.

Can you please revert this back to own?